src/parse.rs

```diff
@@ -556,98 +568,122 @@ where
                     // ClassEscape :: b
                     'b' => {
                         self.consume('b');
-                        Ok(Some(ClassAtom::CodePoint(u32::from('\x08'))))
+                        Ok(Some((ClassAtom::CodePoint(u32::from('\x08')), None)))
                     }
                     // ClassEscape :: [+UnicodeMode] -
                     '-' if self.flags.unicode => {
                         self.consume('-');
-                        Ok(Some(ClassAtom::CodePoint(u32::from('-'))))
+                        Ok(Some((ClassAtom::CodePoint(u32::from('-')), None)))
                     }
-                    // ClassEscape :: [~UnicodeMode] c ClassControlLetter
                     'c' if !self.flags.unicode => {
-                        let input = self.input.clone();
                         self.consume('c');
                         match self.peek().map(to_char_sat) {
+                            // ClassEscape :: [~UnicodeMode] c ClassControlLetter
                             Some('0'..='9' | '_') => {
                                 let next = self.next().expect("char was not next");
-                                Ok(Some(ClassAtom::CodePoint(next & 0x1F)))
+                                Ok(Some((ClassAtom::CodePoint(next & 0x1F), None)))
                             }
-                            // ClassEscape :: CharacterEscape
-                            _ => {
-                                self.input = input;
-                                let cc = self.consume_character_escape()?;
-                                Ok(Some(ClassAtom::CodePoint(cc)))
+                            // CharacterEscape :: c AsciiLetter
+                            Some('a'..='z' | 'A'..='Z') => {
+                                let next = self.next().expect("char was not next");
+                                Ok(Some((ClassAtom::CodePoint(next % 32), None)))
                             }
+                            // ClassAtomNoDash :: \ [lookahead = c]
+                            _ => Ok(Some((
+                                ClassAtom::CodePoint(u32::from('\\')),
+                                Some(ClassAtom::CodePoint(u32::from('c'))),
```

Comment:
I wasn't able to follow this - from what I can [tell in the spec](https://262.ecma-international.org/#sec-patterns-static-semantics-character-value-annexb) it says:

> Return the numeric value of U+005C (REVERSE SOLIDUS).

but the actual behavior in Chrome agrees with your code - do you know where this is documented?

Reply:
I think you can read the spec like you have suggested to write to code in your other comment. It just says that if we encounter a `\` and the next char is a `c` we return `U+005C`. Then the spec `returns` and we read the `c` as a single char next.